controllers/loadtest_controller_test.go

```diff
@@ -502,3 +501,134 @@ var _ = Describe("Pod Creation", func() {
 		})
 	})
 })
+
+var _ = Describe("CheckMissingPods", func() {
+
+	var currentLoadTest = newLoadTestWithMultipleClientsAndServers() // nothing modify this --> okay
+	var expectedReturnList []*grpcv1.Component                       //initialized every time after a test -->okay
+	var allRunningPods = &corev1.PodList{Items: []corev1.Pod{}}
+	var returnedList []*grpcv1.Component // it was a newlist generated form myfun everytime, address are different --> okay
+
+	JustBeforeEach(func() {
+		returnedList = CheckMissingPods(currentLoadTest, allRunningPods)
+	})
+
+	AfterEach(func() {
+		expectedReturnList = []*grpcv1.Component{}
+		newThing := corev1.PodList{Items: []corev1.Pod{}}
+		allRunningPods = &newThing
+	})
+
+	Describe("no pods from current loadtest is running", func() {
+		BeforeEach(func() {
+			//Set up the expected returned list, in these cases should be full list
+			for i := 0; i < len(currentLoadTest.Spec.Clients); i++ {
+				expectedReturnList = append(expectedReturnList, &currentLoadTest.Spec.Clients[i].Component)
+			}
+			for i := 0; i < len(currentLoadTest.Spec.Servers); i++ {
+				expectedReturnList = append(expectedReturnList, &currentLoadTest.Spec.Servers[i].Component)
+			}
+			expectedReturnList = append(expectedReturnList, &currentLoadTest.Spec.Driver.Component)
+		})
+
+		Context("allRunningPods is empty", func() {
+			allRunningPods = &corev1.PodList{Items: []corev1.Pod{}}
+			It("return the full list", func() {
+				Expect(checkIfEqual(returnedList, expectedReturnList)).To(Equal(true))
+			})
+		})
+
+		Context("irrelevant pods are running", func() {
+			allRunningPods = &corev1.PodList{Items: []corev1.Pod{}}
+			allRunningPods.Items = append(allRunningPods.Items, createPodListWithIrrelevantPod().Items...)
+			It("return the full list", func() {
+				Expect(checkIfEqual(returnedList, expectedReturnList)).To(Equal(true))
+			})
+		})
+	})
+
+	Describe("some of pods from current loadtest is running", func() {
+
+		BeforeEach(func() {
+			//Add pod with defaults.ComponentNameLabel: server-1, pod with defaults.ComponentNameLabel: client-2, driver
```

Comment:
This comment could be clearer.

Reply:
I was wondering if I should delete that comment all together, it is self explanatory when look at the codes.